compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -1044,4 +1114,322 @@ mod tests {
         let main = ssa.main();
         assert_eq!(count_loads(main.entry_block(), &main.dfg), 1);
     }
+
+    #[test]
+    fn remove_repeat_loads() {
```

Comment:
We now have a shiny new SSA parser thanks to Ary. Can you update these tests to make use of it?

Reply:
Done. Reduced the diff almost 40% :)